### PR TITLE
feat(codec): Make error when not utf8 value in compression encoding

### PR DESCRIPTION
**Pull request by tottoto**
Makes error in the case when not utf8 value is in the compression encoding.

### CODE REVIEW

**Review by djc — APPROVED** [review]
Good idea!

File: tonic/src/codec/compression.rs [hunk 1]
```diff
@@ -129,26 +129,31 @@ impl CompressionEncoding {
...
+                    Err(_) => {
+                        other_str_tmp = format!("{:?}", other);
+                        &other_str_tmp
+                    }
+                };
+
                 let mut status = Status::unimplemented(format!(
                     "Content is compressed with `{}` which isn't supported",
-                    other
+                    other_str,
```
[the remark this answers is not in the document]
Comment (tottoto) [reply tonic/src/codec/compression.rs]: Updated.

File: tonic/src/codec/compression.rs [hunk 2]
```diff
@@ -129,26 +129,30 @@ impl CompressionEncoding {
...
+                // https://blog.rust-lang.org/2024/06/13/Rust-1.79.0.html#extending-automatic-temporary-lifetime-extension
+                let other_debug_string;
+
                 let mut status = Status::unimplemented(format!(
                     "Content is compressed with `{}` which isn't supported",
-                    other
+                    match std::str::from_utf8(other) {
+                        Ok(s) => s,
+                        Err(_) => {
+                            other_debug_string = format!("{:?}", other);
```
Comment (djc) [inline tonic/src/codec/compression.rs:152]: I actually meant just inlining the `other_str` in the string, but this works, too!

Just now noticing we could also inline the `other` argument, here.

**Review by djc — APPROVED** [review]

File: tonic/src/codec/compression.rs (hunk 2 above)
Comment (tottoto) [reply tonic/src/codec/compression.rs]: Updated.

**Review by djc — APPROVED** [review]

**Pull request merged**